engine/development/deploymentManager/src/main/scala/pl/touk/nussknacker/development/manager/DevelopmentDeploymentManagerProvider.scala

```diff
@@ -0,0 +1,140 @@
+package pl.touk.nussknacker.development.manager
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.api.deployment._
+import pl.touk.nussknacker.engine.api.deployment.simple.{SimpleProcessStateDefinitionManager, SimpleStateStatus}
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.engine.api.queryablestate.QueryableClient
+import pl.touk.nussknacker.engine.api.test.TestData
+import pl.touk.nussknacker.engine.api.{ProcessVersion, StreamMetaData}
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.deployment.{DeploymentData, ExternalDeploymentId, User}
+import pl.touk.nussknacker.engine.testmode.TestProcess
+import pl.touk.nussknacker.engine.{BaseModelData, DeploymentManagerProvider, TypeSpecificInitialData}
+import sttp.client.{NothingT, SttpBackend}
+
+import java.util.UUID
+import java.util.concurrent.{ThreadLocalRandom, TimeUnit}
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+
+class DevelopmentDeploymentManager(actorSystem: ActorSystem) extends DeploymentManager with LazyLogging {
+
+  private val MinSleepTime = 5
+  private val MaxSleepTime = 12
+
+  private val memory: TrieMap[ProcessName, ProcessState] = TrieMap[ProcessName, ProcessState]()
+  private val random: ThreadLocalRandom = ThreadLocalRandom.current()
+
+  implicit private class ProcessStateExpandable(processState: ProcessState) {
+    def withStateStatus(stateStatus: StateStatus): ProcessState = {
```

Comment:
We already have `ProcessState.withStatusDetails`. Can't it be used here? (just asking)